packages/stitch/tests/stitchSchemas.test.ts

```diff
@@ -2147,6 +2153,248 @@ fragment BookingFragment on Booking {
+                                  selectionSet: {
+                                    kind: 'SelectionSet',
+                                    selections: [
+                                      {
+                                        kind: 'Field',
+                                        alias: undefined,
+                                        name: { kind: 'Name', value: 'items' },
+                                        arguments: [],
+                                        directives: [],
+                                        selectionSet: pageSelectionSet.selectionSet,
+                                      }
+                                    ]
+                                  },
+                              }
+                            ]
+
+                            /**
+                             * The is the query after:
+                               {
+                                  persona {
+                                    transactions {
+                                      items {
+                                        id
+                                        debt {
+                                          ...DebtFields
+                                        }
+                                      }
+                                    }
+                                  }
+                                }
+                             */
+
+                            return ast;
+                          }
+                        }
+                      ]
+                    },
+                  );
+
+                  expect(result.transactions.items.length).toBe(2);
+
+                  // Here, we have an issue, we have `id` but no `debt` here
+                  expect(result.transactions.items[0].debt).toBeDefined();
+
+                return {
+                  page: result.transactions.items,
+                  totalCount: result.transactions.items.length
+                };
+              }
+            }
+          },
+        });
+
+        const result = await execute({
+          schema: mergedSchema,
+          contextValue: {},
+          rootValue: {},
+          variableValues: {},
+          document: parse(`
+            query {
+              flattenedTransactions {
+                page {
+                  id
+                  debt {
+                    ...DebtFields
+                  }
+                }
+                totalCount
+              }
+            }
+
+            fragment DebtFields on Debt {
+              debtInstallmentPlan: installmentPlan {
+                category
+                installments
+              }
+            }
+            `,
+            )
+        });
+
+        expect(result.errors).toBeUndefined();
+        expect(result.data).toBeDefined();
+        expect(result.data.flattenedTransactions.page.length).toBe(2);
+        expect(result.data.flattenedTransactions.page[1].debt).toBeDefined();
```

Comment:
Maybe `.not.toBeNull();` instead?

     expect(result.data.flattenedTransactions.page[1].debt).not.toBeNull();

Reply:
Fixed it up.